### DISCUSSION TITLE
An Example of Shortening R Code | /en/2023/10/shorten-code/

### DISCUSSION THREAD

**Discussion by giscus** [rbind/yihui#1526] [Comments]
While I was answering a question from a user last week, I saw a piece of code that, in my opinion, could be shortened. What it does is to set two chunk options, message and warning, to NA or FALSE &hellip;

https://yihui.org/en/2023/10/shorten-code/

**Comment by eliocamp** [comment]
If I recall correctly, `installed.packages()` currently navigates the whole folder structure in your libraries, so it's relatively slow; and painfully slow for large package libraries, so that's another plus of not testing if evaluate is installed. (I think the "correct" way of testing if a package is installed is `requireNamespace("evaluate", quietly = TRUE)`; at least that's what I've seen in other packages and what I use in mine).

**Reply by yihui** [reply]
You are right that `installed.packages()` can be slow.

If we only test for the existence of a package but don't care if it can actually be loaded, I think `system.file(package = 'evaluate') != ''` is a faster way to go.
